### PR TITLE
Update connecting.md with async/await

### PR DESCRIPTION
**Pull request by milaniliev**
`co.js` uses ES6 generators to achieve coroutines, of which `async/await` is a natural version. 

This simplifies the code significantly, removes reliance on `co.js`, removes the need to understand generators, and is compatible with current Node.js and on by default.

### CODE REVIEW

File: docs/reference/content/reference/ecmascript6/connecting.md [hunk 1]
```diff
@@ -30,3 +30,27 @@ co(function*() {
...
+Current Node.js allow usage of async/await syntax, simplifying the above coroutine/generator flow further:
+
+```js
+let MongoClient = require('mongodb').MongoClient
+let asset = require('asset')
+
+let connect = async function(){
+  try {
+    let db = await MongoClient.connect('mongodb://localhost:27017/myproject')
+    db.close()
```
Comment (mbroadst) [inline docs/reference/content/reference/ecmascript6/connecting.md]: `await db.close();`

**Comment by FireController1847** [comment]
Don't async functions always return a promise no matter what? This would mean you'd have to, at base level, run `await connect()`, which is impossible (afaik) since you can't use await on base level.

**Comment by milaniliev** [comment]
@FireController1847 You're right, it's not possible to call `await` at top-level, and async functions always return a promise. So you have two options: you can handle promises without `await`, e.g. with `.then()`, or you can wrap them in a function whose return value you don't care about - that's what `connect()` is, in fact. The top-level simply calls `connect()` and does not need to wait for it to complete or use its return value.

**Comment by FireController1847** [comment]
@milaniliev That's the only issue I had with this PR; While this is literally an example of await and async, this example would not actually work when executing it in the real world unless all of your database code was within the connect() function.

**Comment by dcharbonnier** [comment]
@FireController1847 you can say that for all code using await/async it's not an issue. You also need to execute all your code in a callback with the other method.
it's pretty common to see a 
```
(async () => {
})();
``` 
at the first level of the code, the above code work with the current version of mongodb-native

```
import {MongoClient} from "mongodb";
(async () => {
    const client = new MongoClient();
    const url = 'mongodb://localhost:27017/test';
    const db = await client.connect(url);
    console.log(await db.admin().listDatabases());
    await db.close();
})();
```

**Comment by FireController1847** [comment]
Well if we have to create a function and call it in the first place anyways, then why even use async/await to connect? It'd take _more_ lines to do it this way than it would the .then() way.

This way
```js
(async () => {
  try {
    var db = await (new MongoClient()).connect(url);
  } catch(e) {
    // Handle Error
  }
})();
```

Callback way
```js
(new MongoClient()).connect(url).then(() => {
}).catch(e => {
  // Handle Error
});
```

**Comment by milaniliev** [comment]
You’re right, for this tiny example, async/await is more code. What I was
trying to demonstrate is how to use async/await, so that users can use it
in other places, or with more complex connections/queries, where the
async/await approach involves much less code and is significantly simpler
to read and comprehend, and the code is already inside a function so
there’s no need to wrap it.

I agree that this is probably the most laborious place to demonstrate
async/await use. I’m trying to find a better place for it in the docs
that’s still early enough in the docs reading path that relatively new
readers don’t have to get bogged down in .then() promise chains before
seeing there’s a cleaner, more modern way to do the same.

What do you think?

**Comment by FireController1847** [comment]
@milaniliev I think, personally, the example should be more obvious in the fact that it should be used in deep code or complex situations rather than it seeming like it's possible to use this at base level. I would say try adding some random code around it, or putting comments to specify that this is not meant to be used at base level (because for those who don't know much about async/await will most likely take it that way).

File: docs/reference/content/reference/ecmascript6/connecting.md [hunk 2]
```diff
@@ -30,3 +30,27 @@ co(function*() {
...
+
+# async/await
+
+Current Node.js allow usage of async/await syntax, simplifying the above coroutine/generator flow further:
+
+```js
+let MongoClient = require('mongodb').MongoClient
+let asset = require('asset')
+
+let connect = async function(){
```
Comment (daprahamian) [inline docs/reference/content/reference/ecmascript6/connecting.md]: can we instead just have this be an anonymous async IIFE?

```js
(async function() {
  try {
    let db = await MongoClient.connect('mongodb://localhost:27017/myproject')
    await db.close();
  } catch(error) {
    console.log(error.stack);
  }
})();
```

**Comment by julien-c** [comment]
FWIW top-level async await is coming to Node soon (and has been shipped in v8 for almost a year)

**Comment by daprahamian** [comment]
The documentation is undergoing a complete rewrite, and this PR has not seen activity in over a year, so closing this.

**Pull request closed without merging**